src/js/resolver.js

```diff
@@ -14,9 +14,28 @@ Resolver.prototype.resolve = function(spec, callback, scope) {
   // models
   for(name in spec.definitions) {
     var model = spec.definitions[name];
-    for(propertyName in model.properties) {
-      property = model.properties[propertyName];
-      this.resolveTo(property, resolutionTable);
+    if (!model.properties) {
+      model.properties = {};
+    }
+    if (model.allOf) {
+      for(mergeName in model.allOf){
```

Comment:
Instead of `for(mergeName in model.allOf){  merge = model.allOf[mergeName];` use underscore: `_.values(model.allOf)`

`for..in` is not safe to use. It might iterate in prototype of the object too. 
in line 15 we use `for..in` again. I wish we could fix that too.